src/vs/editor/contrib/linesOperations/common/linesOperations.ts

```diff
@@ -401,3 +407,267 @@ export class DeleteAllLeftAction extends EditorAction {
+
+	public run(accessor: ServicesAccessor, editor: ICommonCodeEditor): TPromise<void> {
+		const originalCursorPosition = editor.getSelection().getStartPosition();
+
+		// 1. paste
+		editor.focus();
+		document.execCommand('paste');
+
+		const currentCursorPosition = editor.getSelection().getStartPosition();
+		const pastedContentRange = new Selection(currentCursorPosition.lineNumber, currentCursorPosition.column, originalCursorPosition.lineNumber, originalCursorPosition.column);
+
+		// 2. format
+		const model = editor.getModel();
+		const { tabSize, insertSpaces } = model.getOptions();
+		const workerService = accessor.get(IEditorWorkerService);
+		const formattingPromise = getDocumentRangeFormattingEdits(model, pastedContentRange, { tabSize, insertSpaces });
+
+		if (!formattingPromise) {
+			return TPromise.as(void 0);
+		}
+
+		const state = editor.captureState(CodeEditorStateFlag.Value, CodeEditorStateFlag.Position);
+
+		return formattingPromise.then(edits => workerService.computeMoreMinimalEdits(editor.getModel().uri, edits, editor.getSelections())).then(edits => {
+			if (!state.validate(editor) || isFalsyOrEmpty(edits)) {
+				return;
+			}
+
+			const command = new EditOperationsCommand(edits, editor.getSelection());
+			editor.executeCommand(this.id, command);
+			editor.focus();
+
+			// 3. update cursor positions
+			let selection = editor.getSelection();
+			selection = new Selection(selection.endLineNumber, selection.endColumn, selection.endLineNumber, selection.endColumn);
+			editor.setSelection(selection);
+		});
+	}
+}
+
+@editorAction
+export class UpperCaseAction extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.transformToUppercase',
+			label: nls.localize('editor.transformToUppercase', "Transform to Uppercase"),
+			alias: 'Transform to Uppercase',
+			precondition: EditorContextKeys.Writable
+		});
+	}
+
+	public run(accessor: ServicesAccessor, editor: ICommonCodeEditor): void {
+		let selections = editor.getSelections();
+		let model = editor.getModel();
+		let commands: ICommand[] = [];
+
+		for (let i = 0, len = selections.length; i < len; i++) {
+			let selection = selections[i];
+			if (selection.isEmpty()) {
+				let cursor = selection.getStartPosition();
+				let word = model.getWordAtPosition(cursor);
+				let wordRange = new Range(cursor.lineNumber, word.startColumn, cursor.lineNumber, word.endColumn);
+
+				if (wordRange !== undefined) {
+					let text = model.getValueInRange(wordRange);
+					commands.push(new ReplaceCommandThatPreservesSelection(wordRange, text.toLocaleUpperCase(),
+						new Selection(cursor.lineNumber, cursor.column, cursor.lineNumber, cursor.column)));
+				}
+			} else {
+				let text = model.getValueInRange(selection);
+				commands.push(new ReplaceCommandThatPreservesSelection(selection, text.toLocaleUpperCase(), selection));
+			}
+		}
+
+		editor.executeCommands(this.id, commands);
+	}
+}
+
+@editorAction
+export class LowerCaseAction extends EditorAction {
```

Comment:
I suggest to create an abstract `CaseAction` that has an abstract method `_modifyText` or whatever that can be overwritten to use `toLocaleUpperCase` and `toLocalLowerCase` to avoid code duplication.